src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -0,0 +1,376 @@
+    :param max_generated_tokens: the maximum number of tokens to generate
+        given the input sequence. If None, the model will generate
+        tokens until the end of the sequence is reached.
+        Otherwise, it will generate up to the maximum number of tokens or end of
+        sequence is reached.
+    :param prompt_processing_sequence_length: For large prompts, the prompt is
+        processed in chunks of this length. This is to maximize the inference
+        speed. By default, this is set to 128.
+    :param force_max_tokens: if True, the pipeline will generate the maximum number
+        of tokens supplied even if the stop token is reached.
+    :param use_deepsparse_cache: if True, the pipeline will use the deepsparse kv cache
+        for caching the model outputs.
+    :param kwargs: kwargs to pass to the TransformersPipeline
+    """
+
+    def __init__(
+        self,
+        deterministic: bool = True,
+        sampling_temperature: float = 1.0,
+        max_generated_tokens: Optional[int] = 1024,
+        # TODO: Set this to 64 once we modify the OPT injection logic
+        prompt_processing_sequence_length: int = 128,
+        force_max_tokens: bool = False,
+        use_deepsparse_cache: bool = False,
+        **kwargs,
+    ):
+        if use_deepsparse_cache:
+            if kwargs["engine_type"] != DEEPSPARSE_ENGINE:
+                raise ValueError(
+                    "`use_deepsparse_cache` is set to True "
+                    "but the chosen `engine_type` "
+                    f"is {kwargs['engine_type']}. "
+                    f"Make sure to set `engine_type` to {DEEPSPARSE_ENGINE}"
+                )
+            raise NotImplementedError(
+                "The deepsparse kv cache is not yet "
+                "supported for text generation pipelines"
+            )
+
+        super().__init__(
+            **kwargs, _delay_engine_initialize=True, _delay_overwriting_inputs=True
+        )
+
+        self.deterministic = deterministic
+        self.sampling_temperature = sampling_temperature
+        self.max_generated_tokens = max_generated_tokens
+        self.prompt_processing_sequence_length = prompt_processing_sequence_length
+        self.force_max_tokens = force_max_tokens
+
+        # override tokenizer to pad to left
+        self.tokenizer.padding_side = "left"
+
+        self.engine = None
+        self.multitoken_engine = NLDecoderEngine(
+            onnx_file_path=self.onnx_file_path,
+            engine_type=self.engine_type,
+            engine_args=self.engine_args,
+            engine_context=self.context,
+            sampling_temperature=self.sampling_temperature,
+            deterministic=self.deterministic,
+            sequence_length=self.sequence_length,
+            input_ids_length=prompt_processing_sequence_length,
+            tokenizer=self.tokenizer,
+            use_deepsparse_cache=use_deepsparse_cache,
+        )
+
+        if self.multitoken_engine.kv_cache_enabled:
```

Comment:
it seems like we're using KV cache enabled as a proxy for if we want to generate multiple tokens or not, no? seems like we might want to make this a separate, explicit control

Reply:
No, we are using 'kv_cache_enabled' if we want to run a simple inference when we map a sequence of tokens to logits.